### PR TITLE
DDF-2984 Add ability to add / remove attributes from results

### PR DESCRIPTION
**Pull request by andrewkfiedler**
#### What does this PR do?
Adds the ability to add / remove attributes from results.  Some parts are still being refactored at the moment, but the feature is complete.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@andrewkfiedler
@jlcsmith 
@rzwiefel 
@clockard 

#### How should this be tested? (List steps with links to updated documentation)
Select a result and try adding and removing attributes.  Try this with multiple results selected.  Verify it won't allow adding / removing in scenarios it shouldn't (editing turned off, revisions, archived, etc.).

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2984

### CODE REVIEW

**Comment by clockard** [comment]
The Hide/Show and Rearrange options in the actions drop down don't seem to do anything when selected. 
screen shot 2017-04-12 at 12 23 29 pm

**Comment by rzwiefel** [comment]
build now

File: catalog/ui/catalog-ui-search/src/main/webapp/js/Common.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ define([
     var format = 'DD MMM YYYY HH:mm:ss.SSS';
 
     return {
+        //randomly generated guid guaranteed to be unique ;)
+        undefined: '2686dcb5-7578-4957-974d-aaa9289cd2f0',
```
Comment (andrewkfiedler) [inline catalog/ui/catalog-ui-search/src/main/webapp/js/Common.js]: swiftexit

Comment (andrewkfiedler) [inline catalog/ui/catalog-ui-search/src/main/webapp/js/Common.js]: But really, this is used to handle the need for a unique key in a map that can technically have any value as a key and also needs to be able to support blank as well as undefined (as two distinct keys).

**Comment by andrewkfiedler** [comment]
build now

**Comment by andrewkfiedler** [comment]
build now

**Comment by andrewkfiedler** [comment]
build now

**Pull request merged**